sherpa/sim/mhtest.py

```diff
@@ -0,0 +1,589 @@
+import numpy as np
+
+from sherpa.sim.mh import dmvnorm
+from sherpa.utils.err import ParameterErr
+from sherpa.astro.io import io_opt
+from sherpa.utils import NoNewAttributesAfterInit, print_fields
+from sherpa.stats import Cash, CStat, WStat
+from sherpa.astro import ui
+
+if io_opt == 'crates_backend':
+    from pycrates import read_file, get_keyval, copy_colvals, get_col_names
+else:
```

Comment:
this needs to be something like

```
elif io_opt == "pyfits_backend":
    from astropy.io import fits
else:
    do we want to error out here or just create a warning message?
```

Ideally we'd be using the IO library to access the data rather than call through to the backend directly. I think we can do this with sherpa.astro.io.read_table_blocks (and hence get rid of this and the backend-specific PCA sub-classes below.

It should be something like

```
# note: no error checking
indata = sherpa.astro.io.read_table_blocks(filename)
hdr = indata[2][1]

mypca = PCA(infile)
mypca.emethod = hdr["EMETHOD"]
mypca.cmethod = hdr["CMETHOD"]
mypca.errext = hdr["ERREXT"]
mypca.bias = indata[1][1]["BIAS"]

block = block[1][2]
mypca.colnames = list(block.keys())  # this should work
mypca.component = block["COMPONENT"]
mypca.fvariance = block["FVARIANCE"]
mypca.eigenval = block["EIGENVAL"]
mypca.eigenvec = block["EIGENVEC"]
```